### ISSUE TITLE
Add W&B experiment tracking to training example

### ISSUE THREAD

**Issue by evgri243** [JetBrains-Research/opaque#49]
## Summary
Add optional Weights & Biases integration to `train_causal_lm.py` for experiment tracking. The company has W&B available, but it should NOT be a hard dependency.

## Implementation
1. `--wandb` flag enables W&B logging (default: disabled)
2. `--wandb-project` and `--wandb-entity` optional args
3. Guard with `try: import wandb` — skip gracefully if not installed
4. Log per step: loss, gradient norm, clip rate, learning rate
5. Log per epoch: epsilon, delta, advantage, clip_norm (if adaptive)
6. Log config: all CLI args as W&B config

## What NOT to Do
- Do NOT add `wandb` to any dependency group
- Do NOT require W&B for any functionality
- Do NOT log model weights or gradients to W&B (privacy risk!)

## Example Usage
```bash
# Without W&B (default)
python train_causal_lm.py --model_name gpt2

# With W&B
pip install wandb  # user installs separately
python train_causal_lm.py --model_name gpt2 --wandb --wandb-project opaque-experiments
```

## Acceptance Criteria
- [ ] W&B integration is fully optional
- [ ] `--wandb` flag with clear error if wandb not installed
- [ ] Training metrics logged per step
- [ ] Privacy metrics logged per epoch
- [ ] Config logged as hyperparameters
- [ ] No W&B dependency in pyproject.toml

## Parent
- #65
